mypy/checker.py

```diff
@@ -1868,6 +1868,19 @@ def check_method_override_for_base_with_name(
                 original_class_or_static = False  # a variable can't be class or static
 
             if isinstance(original_type, FunctionLike):
+                active_self_type = self.scope.active_self_type()
+                if isinstance(original_type, Overloaded) and active_self_type:
+                    # If we have an overload, filter to overloads that match the self type.
+                    # This avoids false positives for concrete subclasses of generic classes,
+                    # see testSelfTypeOverrideCompatibility for an example.
+                    # It's possible we might want to do this as part of bind_and_map_method
+                    filtered_items = [
+                        item
+                        for item in original_type.items
+                        if not item.arg_types or is_subtype(active_self_type, item.arg_types[0])
+                    ]
+                    if filtered_items:
```

Comment:
If there are no filtered_items, shouldn't we always treat the override as compatible? e.g. in your `F` test below, I think the override is actually compatible: the base class says nothing about what the method should return if self is an `A[bytes]`.

Reply:
Yeah, I wasn't sure about this, but it seemed like a weird situation so I chose to fail. Feels sketchy to introduce a `Callable[..., Any]` ... I'll add a comment and if it ever comes up we can reconsider